src/chain.h

```diff
@@ -459,21 +469,38 @@ class CDiskBlockIndex : public CBlockIndex
     bool isStakedAndNotaryPay() const;
     bool isStakedAndAfterDec2019(unsigned int nTime) const;
 
+    //! This method should not be called on a CDiskBlockIndex.
+    void TrimSolution()
+    {
+        assert(!"called CDiskBlockIndex::TrimSolution");
```

Comment:
This is a bit strange.
Why would we need yet another TrimSolution() which just fails, isn't it better not to have it at all and prevent builds with such calls?